### PR TITLE
add UI testing and client explorer

### PR DESCRIPTION
**Pull request by mhost39**
### Related Issues
https://github.com/threefoldtech/js-sdk/issues/1942

### CODE REVIEW

File: tests/frontend/tests/test_settings.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        self.assertIn(email, emails.keys())
+
+        self.info("Delete the email")
+        settings.delete_escalation_emails(email)
+
+        self.info("Check that the email has been deleted from the escalation emails")
+        emails = settings.list("Escalation Emails")
+        self.assertNotIn(email, emails.keys())
+
+    def test04_developer_options(self):
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_settings.py:94]: every button should have a separate test

File: tests/frontend/tests/test_threebot_deployer.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        self.threebot_deployer = ThreebotDeployer(self.driver)
+        self.threebot_deployer.load()
+
+    def tearDown(self):
+        self.wallet_name = self.random_name()
+        self.create_wallet(self.wallet_name)
+        super().tearDown()
+
+    def test01_deploy_and_delete_3bot(self):
+        """
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_threebot_deployer.py:29]: All tests on this module don't follow the shape of our docstring

File: tests/frontend/tests/test_threebot_deployer.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+@pytest.mark.integration
+class ThreebotDeployerTests(BaseTest):
+
+    wallet_name = "wallet_{}".format(randint(1, 500))
+
+    def setUp(self):
+        super().setUp()
+        # if not hasattr(ThreebotDeployerTests, "wallet_name"):
+        self.wallet_name = self.random_name()
+        self.create_wallet(self.wallet_name)
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_threebot_deployer.py:18]: the wallet should be imported as we use one wallet for all tests

File: tests/frontend/tests/test_threebot_deployer.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+        self.threebot_deployer = ThreebotDeployer(self.driver)
+        self.threebot_deployer.load()
+
+    def tearDown(self):
+        self.wallet_name = self.random_name()
+        self.create_wallet(self.wallet_name)
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_threebot_deployer.py:25]: `delete_wallet` instead?

File: tests/frontend/tests/test_threebot_deployer.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+        **Test Scenario**
+        #. Create a wallet.
+        #. Create a threebot instance.
+        #. Check that the threebot instance has been created successfully.
+        #. Delete the threebot instance.
+        #. Check that the threebot instance has been deleted successfully.
+        #. Delete the wallet.
+        """
+        self.info("Create a wallet")
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_threebot_deployer.py:40]: I think it should be import wallet and the log should be in `setUpClass`

File: tests/frontend/tests/test_threebot_deployer.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        self.assertIn(threebot_name, my_3bot_instances)
+
+        self.info("Delete the threebot instance")
+        self.threebot_deployer.delete_threebot_instance(my_3bot_instances=threebot_name, password=password)
+
+        self.info("Check that the threebot instance has been deleted successfully")
+        my_3bot_instances = self.threebot_deployer.view_my_3bot("DELETED")
+        self.assertIn(threebot_name, my_3bot_instances)
+
+        self.info("Delete the wallet")
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_threebot_deployer.py:62]: this log should be in `tearDownClass`

File: tests/frontend/tests/test_threebot_deployer.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+from tests.frontend.pages.wallets.wallets import Wallets
+from tests.frontend.pages.threebot_deployer.threebot_deployer import ThreebotDeployer
+
+
+@pytest.mark.integration
+class ThreebotDeployerTests(BaseTest):
+
+    wallet_name = "wallet_{}".format(randint(1, 500))
+
+    def setUp(self):
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_threebot_deployer.py:14]: let's use `setUpClass` and `tearDownClass` for these tests.

File: tests/frontend/tests/test_threebot_deployer.py [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+        #. Create a threebot instance.
+        #. Stopped the new created 3bot instance.
+        #. Check that the 3bot instance has been stopped successfully.
+        #. Start the 3bot instance.
+        #. Check that the 3bot instance has been started successfully.
+        #. Delete the 3bot instance.
+        #. Delete the wallet.
+        """
+
+        self.info("Create a wallet")
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_threebot_deployer.py:79]: the same

File: tests/frontend/tests/test_threebot_deployer.py [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+        self.info("Check that the 3bot instance has been started successfully")
+        self.assertEqual(urllib.request.urlopen(threebot_instance_url).getcode(), 200)
+        my_3bot_instances = self.threebot_deployer.view_my_3bot("RUNNING")
+        self.assertIn(threebot_name, my_3bot_instances)
+
+        self.info("Delete the 3bot instance")
+        self.threebot_deployer.delete_threebot_instance(my_3bot_instances=threebot_name, password=password)
+
+        self.info("Delete the wallet")
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_threebot_deployer.py:108]: the same

File: tests/frontend/tests/test_threebot_deployer.py [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+        self.info("Create a wallet")
+        wallets = Wallets(self.driver)
+        wallets.load()
+        wallets.add_funded(wallet_name)
+
+    def delete_wallet(self, wallet_name):
+        self.info("Delete the wallet")
+        wallets = Wallets(self.driver)
+        wallets.load()
+        wallets.delete(wallet_name)
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_threebot_deployer.py:120]: this should be removed after importing the wallet in `setUpClass`

File: tests/frontend/tests/test_workloads.py [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+@pytest.mark.integration
+class WorkloadsTests(BaseTest):
+    def setUp(self):
+        super().setUp()
+        self.workloads = workloads(self.driver)
+        self.workloads.load()
+
+    def test01_delete_selected_workloads(self):
+        """
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_workloads.py:15]: this test doesn't follow the shape of our docstring

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
- Please generate the docs of these tests

File: tests/frontend/tests/test_threebot_deployer.py [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.info("Import a wallet")
+        cls.wallet_name = "demos_wallet"
+        cls.wallet_secret = os.environ.get("WALLET_SECRET")
+        if not cls.wallet_secret:
+            raise Exception("Please add WALLET_SECRET as environment variables")
+
+        j.clients.stellar.get(cls.wallet_name, network="STD", secret=cls.wallet_secret)
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_threebot_deployer.py:23]: Please get the wallet then change the secret as we do in vdc tests https://github.com/threefoldtech/js-sdk/blob/development/tests/sals/vdc/vdc_base.py#L37

File: tests/frontend/tests/test_threebot_deployer.py [hunk 13]
```diff
@@ -0,0 +1,130 @@
+import pytest
+import urllib.request
+import os
+from jumpscale.loader import j
+from random import randint
+from tests.frontend.tests.base_tests import BaseTest
+from tests.frontend.pages.wallets.wallets import Wallets
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_threebot_deployer.py:7]: not needed?

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
Please add a screenshot for running these tests

**Comment by mhost39** [comment]
settings: 
image

explorer:
image

workloads:
image

File: tests/frontend/tests/test_settings.py [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+        self.assertIn(admin, admins.keys())
+
+        self.info("Delete the admin")
+        settings.delete_admin(admin)
+
+        self.info("Check that the admin has been deleted from the admins")
+        admins = settings.list("Admins")
+        self.assertNotIn(admin, admins.keys())
+
+    @pytest.mark.skip("can't get inputs name")
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_settings.py:38]: Please open an issue and skip it with the issue link to be tracked.

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
Please resolve the conflict in this branch,

**Pull request closed without merging**